### PR TITLE
[CI] Add design rule which enforces all validate_data inputs in get_namespace when array api enabled

### PR DESCRIPTION
**Pull request by icfaust**
## Description

This adds a rule which parses the trace to verify that the inputs to ```get_namepace``` exist in ```validate_data``` and possibly ```_check_sample_weight```.  ```get_namespace``` verifies that all inputs are of the same array type, which is important for setting the return values and estimator attributes properly.

It skips unless the estimator is array API enabled, as otherwise the input is converted to numpy via ```validate_data```
---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.
- [ ] I have provided justification why quality metrics have changed or why changes are not expected.
- [ ] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
I realize that this is adding to an existing system, but there's a more idiomatic way to construct these tests that would leave more understandable error messages and better traces:
https://pytest-mock.readthedocs.io/en/latest/usage.html#spy

**Comment by david-cortes-intel** [comment]
@icfaust I'm unable to make the test here run when running 'test_common.py'. What would be required to make this test execute and reach the section with the assertion about the inputs?

File: sklearnex/tests/test_common.py [hunk 1]
```diff
@@ -545,11 +547,92 @@ def fit_check_before_support_check(text, estimator, method):
...
+        get_valid_inputs = "".join(validate_data_call.split()).split(",")
+
+        assert "self" == get_valid_inputs[0]
+        for idx, inp in enumerate(get_valid_inputs[1:]):
+            # this is specifically written for validate_data definition and
+            # how we lint sklearnex. Generally X and y should be passed not
+            # as keyword arguments. See:
+            # https://scikit-learn.org/stable/modules/generated/sklearn.utils.validation.validate_data.html
+            if ("=" not in inp and idx < 2) or inp.startswith(("X=", "y=")):
+                assert (
```
Comment (david-cortes-intel) [inline sklearnex/tests/test_common.py:615]: @icfaust Could you add an intentional failure right here to verify that at least one CI job is able to reach this section?

Comment (icfaust) [reply sklearnex/tests/test_common.py:615]: The reason it doesn't run is because none of the estimators do native array api dispatching yet.  This ```get_namespace``` type issue is only relevant once we start introducing that into the estimators.  Its a bit of a chicken and the egg problem.

Comment (david-cortes-intel) [reply sklearnex/tests/test_common.py:615]: Then I guess it'd be better to postpone this PR until we have some estimator with native array API dispatching.

File: daal4py/sklearn/linear_model/logistic_path.py [hunk 2]
```diff
@@ -902,75 +857,35 @@ def daal4py_predict(self, X, resultsToEvaluate):
...
-        dual=dual,
-        penalty=penalty,
-        intercept_scaling=intercept_scaling,
-        multi_class=multi_class,
-        random_state=random_state,
-        check_input=check_input,
-        max_squared_sum=max_squared_sum,
-        sample_weight=sample_weight,
-        l1_ratio=l1_ratio,
+def logistic_regression_path(*args, **kwargs):
```
Comment (david-cortes-intel) [inline daal4py/sklearn/linear_model/logistic_path.py:860]: Since this is an internal function, I think there would be no harm in leaving it with an extra unused argument instead of splitting the signature like this.
